### PR TITLE
Refactor kubernetes scheduler to use statefulsets.

### PR DESCRIPTION
**Pull request by cckellogg**

### CODE REVIEW

**Comment by cckellogg** [comment]
@jerrypeng will you review this too

File: deploy/kubernetes/minikube/apiserver.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ spec:
       serviceAccountName: heron-apiserver
       containers:
         - name: heron-apiserver
-          image: heron/heron:latest
+          image: heron/heron:lastest
```
Comment (srkukarni) [inline deploy/kubernetes/minikube/apiserver.yaml]: What is this?

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java [hunk 2]
```diff
@@ -179,19 +179,37 @@ private SchedulerUtils() {
...
+   * @param shardId the executor/container index
    * @param ports a map of ports to use where the key indicate the port type and the
    * value is the port
    * @return String[] representing the command to start heron-executor
    */
   public static String[] getExecutorCommand(
       Config config,
       Config runtime,
-      int containerIndex,
+      int shardId,
```
Comment (srkukarni) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java]: Can this be done better?

Comment (cckellogg) [reply heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java]: The flag we set is called shard so i updated the variable name to match. We can probably do this better but that is probably a bigger change that we can address in another pr.

**Pull request merged**